sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py

```diff
@@ -56,6 +56,36 @@ def on_request(self, request: PipelineRequest) -> None:
         request.http_request.headers["Ocp-Apim-Subscription-Key"] = self.credential.key
         request.http_request.headers["Ocp-Apim-Subscription-Region"] = self.credential.region
 
+class TranslatorAADCredential:
+    """Credential for Translator Service when using AAD authentication.
+    :param tokenCredential: An object which can provide an access token for the Translator Resource, such as a credential from
+        :mod:`azure.identity`
+    :type tokenCredential: ~azure.core.credentials.TokenCredential
+    :param resourceId: Azure Resource Id of the Translation Resource.
```

Comment:
We need to specify a type
```suggestion
    :param str resourceId: Azure Resource Id of the Translation Resource.
```